OpenRA.Mods.Common/Traits/GivesBounty.cs

```diff
@@ -56,6 +56,9 @@ int GetBountyValue(Actor self)
 
 		int GetDisplayedBountyValue(Actor self)
 		{
+			if (self.IsDead)
```

Comment:
As far as I can tell, this breaks bounty display completely (the initial call to `GetDisplayedBountyValue` happens in `INotifyKilled.Killed`, at which point `Actor.IsDead` should already return `true` for that actor).
The real problem here is that `Cargo` seems to occasionally kill/clear passengers before `GetDisplayedBountyValue` is run on them.

One possible solution would be to cache the total cargo bounty in `Cargo` on `INotifyKilled.Killed` *before* the passengers are killed, and let `GivesBountry` then just take that cached value.

Reply:
I'd prefer it if we could find a solution that didn't require `Cargo` to know about `GivesBounty`.

Reply:
I think the best approach would be to make `GivesBounty` implement `INotifyPassengerEntered` and `INotifyPassengerExited` so that it can maintain its own `Dictionary<Actor, GivesBounty[]> passengerBounties` to use on death.  See `AttackGarrisoned` for a more detailed example to work from.

Reply:
I'm not sure how good this will work with passengers in passengers, but I'll give it a shot.